### PR TITLE
Some minor overhauls

### PR DESCRIPTION
**Pull request by richteer**
See the commit messages for each.

Important notes:
- tests will probably have to be update
- I wrote these on a plane, there's a good chance they are buggy

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,13 +1,88 @@
...
+		path = '.'
+	else:
+		path = args[0]
+
+	if not os.path.exists(path):
+		os.mkdir(path)
+	else: # Dir does exist, ensure clear
+		if os.listdir(path):
+			r = input("Directory not empty, are you sure you want to init here? [y/N]: ")
+			if not r.lower() in ("y", "yes", "yolo"):
```
Comment (tophj) [inline main.py:37]: +1 to yolo init

**Comment by sjrct** [comment]
Update the configs in the `*container` and the `irc` agent.

File: halibot/loader.py [hunk 2]
```diff
@@ -28,31 +28,58 @@ def _find(self, name):
...
 
-		for objname, obj in inspect.getmembers(pymod):
-			if inspect.isclass(obj) and issubclass(obj, self.superclass) and objname != supername:
-				# TODO handle multiple superclasses per python module somehow?
-				self.items[name] = obj
-				return
+		try:
+			obj = getattr(pymod,clsname)
+		except:
+			self.warning("Module '{}' failed to load: Could not find class '{}' in '{}'".format(name, src, cls))
```
Comment (sjrct) [inline halibot/loader.py]: I think this should be an error.

File: halibot/loader.py [hunk 3]
```diff
@@ -28,31 +28,58 @@ def _find(self, name):
...
 		supername = self.superclass.__name__
 
 		with open(basepath + "config.json", "r") as f:
-			# TODO something more with this
 			config = json.loads(f.read())
 
-		src = basepath + config["main"]
+		# Check if using the new loader format or not
+		if not config.get("main"):
+			self.log.warning("Failed to load '{}', 'main' key not found in module config".format(name))
```
Comment (sjrct) [inline halibot/loader.py]: Should be an error.

**Comment by sjrct** [comment]
Looks pretty mergable after that.

**Pull request merged**